src/lib/Tooltip/Tooltip.svelte

```
@@ -18,6 +18,42 @@
+    if (position === 'top' || position === 'bottom') {
+      const overflowRight =
+        bubbleRect.right - alreadyShifted - (window.innerWidth - VIEWPORT_MARGIN);
+      const overflowLeft = VIEWPORT_MARGIN - (bubbleRect.left - alreadyShifted);
+      bubbleShift = overflowRight > 0 ? -overflowRight : overflowLeft > 0 ? overflowLeft : 0;
+    } else {
+      const overflowBottom =
+        bubbleRect.bottom - alreadyShifted - (window.innerHeight - VIEWPORT_MARGIN);
+      const overflowTop = VIEWPORT_MARGIN - (bubbleRect.top - alreadyShifted);
+      bubbleShift = overflowBottom > 0 ? -overflowBottom : overflowTop > 0 ? overflowTop : 0;
+    }
+  });
```

ESLint

[error] $effect is not allowed. Use a different reactive pattern instead.

(no-restricted-syntax)